tools/secret-management/Azure.Sdk.Tools.SecretRotation.Core/RotationPlan.cs

```diff
@@ -102,21 +106,31 @@ public async Task<RotationPlanStatus> GetStatusAsync()
 
             SecretState[] allStates = secondaryStoreStates.Prepend(primaryStoreState).ToArray();
 
-            DateTimeOffset thresholdDate = this.timeProvider.GetCurrentDateTimeOffset().Add(RotationThreshold);
+            DateTimeOffset rotationThresholdDate = this.timeProvider.GetCurrentDateTimeOffset().Add(RotationThreshold);
+
+            DateTimeOffset? warningThresholdDate = WarningThreshold.HasValue
+                ? this.timeProvider.GetCurrentDateTimeOffset().Add(WarningThreshold.Value)
+                : default;
```

Comment:
should we have a smart default? Perhaps half the time of the expire?

Reply:
I think that will be `rotationThreshold / 2`.   for the config:
```json
{
  "rotationPeriod": ".180.00:00:00",
  "rotationThreshold": "30.00:00:00"
}
```

we'd get the implicit:
```json
 "warning": "15.00:00:00"
```

meaning, we create a 180 day secret and rotate it 30 days before expiration.  If we haven't rotated it 15 days before expiration, we start reporting a warning